app/src/main/java/com/example/concordia_campus_guide/adapters/RoutesAdapter.java

```diff
@@ -102,8 +105,9 @@ private void setUIElements(Route route) {
     }
 
     private void setUIWalk(Route route) {
-        mainTransportType.setImageResource(R.drawable.ic_directions_walking);
-
+        if (route.getSteps() != null) {
```

Comment:
why specifically if route.getSteps() is not null?

Reply:
I changed all the checks to check for the summary so it's more consistent and the app is behaving correctly now